### ISSUE TITLE
Incorrect line numbers for C0103

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#119]
Originally reported by: **Leon de Almeida (BitBucket: [leondealmeida](http://bitbucket.org/leondealmeida), GitHub: @leondealmeida?)**

---

The error line is reported to be always one line after the last one of the decorator parameters with relevant data. The code below replicates it.

To validate the bug, add newlines below the `0` (last value on the decorators list). The test case is pretty much absurd - but more complex parameters (think the ones for `pytest.mark.parametrize`) may introduce lists and tuples that will also reproduce this behaviour (which is my original scenario).

```


"""
Some docstring

"""

def dummydecorator(aaa, bbb):
    """Some"""
    def real_decorator(fff):
        """pieces of"""
        def wrapper(*args, **kwargs):
            """docstring"""
            print(aaa, bbb)
            fff(*args, **kwargs)
        return wrapper
    return real_decorator


@dummydecorator(1, [
    0
# C0103 always points here - line 20


])
def a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad():  # line 24
    """Docstring"""
    print('LOL')

a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad()

```

Pylint output (-rn option):

```
#!text

************* Module a
a.py:20:0: C0103: Function name "a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad" doesn't conform to '[a-z_][a-z0-9_]{2,30}$' pattern (invalid-name)
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/119

**Comment by DanielNoord** [comment]
After deprecation of `3.6` this should only be an issue on `3.7`. The ast parser of `3.8` does seem to handle multi line docstrings and decorators on function definitions much better.
We ran into similar issues with https://github.com/PyCQA/astroid/pull/1276

If anybody wants to take a stab at this they should probably look at `astroid` and see if they can come up with a better and consistent way to determine line numbering, although I think the fact that this has been open for 8+ years speaks to the difficulty of this issue..

**Comment by Pierre-Sassoulas** [comment]
> Without a better AST implementation, it's not possible to determine on which line the function actually starts.

Or it's really difficult and will make astroid's code ugly and hard to maintain. I'm closing as won't fix.
